src/public-lib/components/SelectHelper.tsx

```diff
@@ -0,0 +1,21 @@
+import * as React from "react";
+
+export type Option = {
+    value: string;
+    label: string | JSX.Element;
+    disabled?: boolean;
+};
+
+export function getCheckBox(option: Option,
+                            valueMap: {[key: string]: {value: string}}): string
```

Comment:
I think the typing of `valueMap` is confusing - the key is named `key` but you're checking if `option.value` is a key, which makes me feel like the key should be named `value`. Since you're only checking presence, it's confusing to add the type of the values as well. 

In summary, I would call it
`selectedValues: {[ value:string ]: any}`

Reply:
hmm, how about `selectedValues: {[ optionValue:string ]: Partial<Option>}`?

Reply:
yeah but you never use the value, right? so why include it in the typing. you're using it as a set to only test membership

Reply:
ideally this should be a hash set, but I'll go with `any` :)